hathitrust/ht_indexer
# create a subclass to connect to RabbitMQ using the dead letter queue

This PR is about this [task](https://hathitrust.atlassian.net/browse/DEV-1188) and consists of creating a new class to connect to RabbitMQ using a dead letter queue. 

The change of this PR does not introduce a new feature, it is also improving the readability of this repository. 

All the services that use a queue have been updated to utilize the new class and remove the parameter _dead_letter_queue_. Additionally, some tests have been updated. 

In the Dockerfile, the ownership of the /app folder was updated to resolve an issue identified in Kubernetes.
**How to test it?**

1. Clone the repository
`git clone ... or git fetch`

2. Use the branch of this PR
`git checkout -b DEV-1188-queueConnection-subclass`

3. In the working directory,

3.1. Create the image
`docker build -t document_generator .`

3.2. Run the container
 `docker compose up document_generator -d`

3.3. Run Python tests related to document_generator_service
`docker compose exec document_generator pytest document_generator ht_document ht_queue_service ht_utils`

If you see the image below in your terminal, the PR is working fine.

<img width="980" alt="image" src="https://github.com/hathitrust/ht_indexer/assets/47088146/ff3d1860-06d2-4d94-b846-ea8488864b22">
